etcd-backup-operator/pkg/operator/sync.go

```diff
@@ -0,0 +1,67 @@
+package operator
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func (b *Backup) runWorker() {
+	for b.processNextItem() {
+	}
+}
+
+func (b *Backup) processNextItem() bool {
+	// Wait until there is a new item in the working queue
+	key, quit := b.queue.Get()
+	if quit {
+		return false
+	}
+	// Tell the queue that we are done with processing this key. This unblocks the key for other workers
+	// This allows safe parallel processing because two pods with the same key are never processed in
+	// parallel.
+	defer b.queue.Done(key)
+	err := b.processItem(key.(string))
+	// Handle the error if something went wrong during the execution of the business logic
+	b.handleErr(err, key)
+	return true
+}
+
+func (b *Backup) processItem(key string) error {
+	obj, exists, err := b.indexer.GetByKey(key)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+
+	cobj, err := scheme.Scheme.DeepCopy(obj)
+	if err != nil {
+		return err
+	}
+	eb := cobj.(*api.EtcdBackup)
+	b.handleBackup(&eb.Spec)
+	return nil
+}
+
+func (b *Backup) handleBackup(spec *api.EtcdBackupSpec) error {
+	if spec.StorageType != "s3" {
+		return fmt.Errorf("unknown storage type: %v", spec.StorageType)
+	}
+	if spec.BackupStorageSource.S3 == nil {
```

Comment:
This should be written as validation method and checked at the beginning.
It shouldn't return error here because workqueue will retry.

Reply:
how does workqueue retrying work?